podpac/core/cache/hash_cache.py

```diff
@@ -0,0 +1,289 @@
+from podpac.core.node import Node, NodeDefinitionError, NodeException
+from podpac.core.utils import NodeTrait
+from podpac.core.utils import trait_is_defined
+import traitlets as tl
+from podpac.core.coordinates import Coordinates
+from podpac.core.cache import CacheCtrl, get_default_cache_ctrl, make_cache_ctrl, S3CacheStore, DiskCacheStore
+from podpac.core.managers.multi_threading import thread_manager
+from podpac import settings
+from podpac.core.cache.cache_ctrl import _CACHE_STORES
+
+
+class HashCache(Node):
+    """
+    A node that caches the output of another node.
+
+    Attributes
+    ----------
+    source : podpac.Node
+        The source node to cache.
+    cache_ctrl: :class:`podpac.core.cache.cache.CacheCtrl`
+        Class that controls caching. If not provided, uses default based on settings.
+    cache_coordinates : podpac.Coordinate, optional
+        Coordinates that should be used for caching. If not provided. self.source.coordinates will be used, if it exists. Otherwise, use the request coordinates for the cache.
+    _relevant_dimensions : list, optional
+        The relevant dimensions for caching.
+    """
+
+    source = NodeTrait(allow_none=True).tag(attr=True, required=True)  
+    cache_coordinates = tl.Instance(Coordinates, allow_none=True, default_value=None, read_only=True)
+    _relevant_dimensions = tl.Instance(list, allow_none=True, default_value=None)
+    cache_ctrl = tl.Instance(CacheCtrl, allow_none=True)
+    _from_cache = tl.Bool(allow_none=True, default_value=False)
+    cache_type = tl.Union(
+        [tl.List(tl.Enum(_CACHE_STORES.keys())), tl.Enum(_CACHE_STORES.keys())], allow_none=True, default=None
+    )
+
+    def __getattr__(self, name):
```

Comment:
Removing this will throw errors for any attribute not hardcoded into `HashCache` to be equal to `self.source.attribute`. The only way to do this without using __getattr__ is to manually hard-code all of these attributes in.